canonical-web-and-design/snapcraft.io
# Release UI components refactor

Splits release UI table components into smaller, more flexible components.

### Code review

This PR is full of moving core around between files, renaming files and things, etc.
The diff view is likely not going to be very useful. It's probably best to checkout the branch and review current state of affected files. Most of the changes are contained in components/releasesTable folder: https://github.com/bartaz/snapcraft-flask/tree/release-ui-row-refactor/static/js/publisher/release/components/releasesTable

### QA

Almost every component of releases table view was affected by the refactor. It's important to QA all of the release UI functionality, so we are sure there are no regressions.

- ./run or https://snapcraft-io-canonical-web-and-design-pr-2280.run.demo.haus/
- go to releases page of any snap
- check all of the release UI functionality:
  - drag'n'drop of revisions and channels
  - promoting via menu
  - closing channels
  - viewing history
  - branches
  - anything else you can think of...
- try with releases page of different snap (with one or multiple arch, with multiple tracks, with branches, with builds from launchpad...)
